src/sagemaker/jumpstart/model.py

```diff
@@ -390,30 +397,29 @@ def _create_sagemaker_model(
         # inference endpoint.
         if self.model_package_arn and not self._model_data_is_set:
             # When a ModelPackageArn is provided we just create the Model
-            match = re.match(MODEL_PACKAGE_ARN_PATTERN, self.model_package_arn)
-            if match:
-                model_package_name = match.group(3)
-            else:
-                # model_package_arn can be just the name if your account owns the Model Package
-                model_package_name = self.model_package_arn
-            container_def = {"ModelPackageName": self.model_package_arn}
-
-            if self.env != {}:
-                container_def["Environment"] = self.env
-
-            if self.name is None:
-                self._base_name = model_package_name
-
-            self._set_model_name_if_needed()
-
-            self.sagemaker_session.create_model(
-                self.name,
-                self.role,
-                container_def,
+            model_package = ModelPackage(
+                role=self.role,
+                model_data=self.model_data,
+                model_package_arn=self.model_package_arn,
+                sagemaker_session=self.sagemaker_session,
+                predictor_cls=self.predictor_cls,
                 vpc_config=self.vpc_config,
-                enable_network_isolation=self.enable_network_isolation(),
```

Comment:
why not enabling network isolation

Reply:
had removed it since to create the model, this is not getting used


as the `_create_sagemaker_model` of the ModelPackage calls its own `self.enable_network_isolation()` so this value will either way be overwritten.

https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/model.py#L1942